src/main/java/com/faforever/client/fx/WebViewConfigurer.java

```diff
@@ -58,19 +55,13 @@ public void configureWebView(WebView webView) {
       themeService.registerWebView(webView);
 
       ((JSObject) engine.executeScript("window")).setMember(JAVA_REFERENCE_IN_JAVASCRIPT, browserCallback);
-
-      Document document = webView.getEngine().getDocument();
-      if (document == null) {
-        return;
-      }
-
-      NodeList nodeList = document.getElementsByTagName("a");
-      for (int i = 0; i < nodeList.getLength(); i++) {
-        Element link = (Element) nodeList.item(i);
-        String href = link.getAttribute("href");
-
-        link.setAttribute("href", "javascript:java.openUrl('" + href + "');");
-      }
+      engine.executeScript(
+        "document.onclick = function (elt) {" + 
+        "  document.querySelectorAll(\"a[target]\").forEach(e => {" + 
+        "    if (!e.href.includes(\"javascript\")) " + 
+        "      e.href = \"javascript:java.openUrl('\" + e.href + \"')\"" + 
+        "  });" +
+        "}");
```

Comment:
I don't quite understand this function. So everytime the document is clicked it is swapping out the href of all the links in the document right? When someone first clicks a link wouldn't this not work because the onclick event propagates from the inside out? Also this assumes the event did not consume the event and stop propagation right?

Reply:
I wonder if MutationObserver is supported by webview and we would rather use that. https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver

Reply:
Yea your understanding is correct, its not great. it does seem to work on all the links in the news page, even on first click. Yes it assumes the event was not consumed. 

Im happy to explore if mutationobserver will work because I dont like this solution at all, it was a starting point. Another option I explored is finding the href of the clicked element which was infeasible since sometimes its the clicked element, or a direct parent or child, or a descendant of either.